scene/gui/graph_node.h

```diff
@@ -31,170 +31,113 @@
-
-	void set_slot_color_right(int p_idx, const Color &p_color_right);
-	Color get_slot_color_right(int p_idx) const;
-
-	bool is_slot_draw_stylebox(int p_idx) const;
-	void set_slot_draw_stylebox(int p_idx, bool p_enable);
-
 	void set_title(const String &p_title);
 	String get_title() const;
 
-	void set_text_direction(TextDirection p_text_direction);
-	TextDirection get_text_direction() const;
-
-	void set_language(const String &p_language);
-	String get_language() const;
+	HBoxContainer *get_titlebar_hbox();
 
-	void set_position_offset(const Vector2 &p_offset);
-	Vector2 get_position_offset() const;
+	void set_slot(int p_slot_index, bool p_enable_left, int p_type_left, const Color &p_color_left, bool p_enable_right, int p_type_right, const Color &p_color_right, const Ref<Texture2D> &p_custom_left = Ref<Texture2D>(), const Ref<Texture2D> &p_custom_right = Ref<Texture2D>(), bool p_draw_stylebox = true);
+	void clear_slot(int p_slot_index);
+	void clear_all_slots();
 
-	void set_selected(bool p_selected);
-	bool is_selected();
+	bool is_slot_enabled_left(int p_slot_index) const;
+	void set_slot_enabled_left(int p_slot_index, bool p_enable);
 
-	void set_drag(bool p_drag);
-	Vector2 get_drag_from();
+	void set_slot_type_left(int p_slot_index, int p_type);
+	int get_slot_type_left(int p_slot_index) const;
 
-	void set_show_close_button(bool p_enable);
-	bool is_close_button_visible() const;
+	void set_slot_color_left(int p_slot_index, const Color &p_color);
+	Color get_slot_color_left(int p_slot_index) const;
 
-	int get_connection_input_count();
-	int get_connection_input_height(int p_port);
-	Vector2 get_connection_input_position(int p_port);
-	int get_connection_input_type(int p_port);
-	Color get_connection_input_color(int p_port);
-	int get_connection_input_slot(int p_port);
+	bool is_slot_enabled_right(int p_slot_index) const;
+	void set_slot_enabled_right(int p_slot_index, bool p_enable);
 
-	int get_connection_output_count();
-	int get_connection_output_height(int p_port);
-	Vector2 get_connection_output_position(int p_port);
-	int get_connection_output_type(int p_port);
-	Color get_connection_output_color(int p_port);
-	int get_connection_output_slot(int p_port);
+	void set_slot_type_right(int p_slot_index, int p_type);
+	int get_slot_type_right(int p_slot_index) const;
 
-	void set_overlay(Overlay p_overlay);
-	Overlay get_overlay() const;
+	void set_slot_color_right(int p_slot_index, const Color &p_color);
+	Color get_slot_color_right(int p_slot_index) const;
 
-	void set_resizable(bool p_enable);
-	bool is_resizable() const;
+	bool is_slot_draw_stylebox(int p_slot_index) const;
+	void set_slot_draw_stylebox(int p_slot_index, bool p_enable);
 
-	void set_draggable(bool p_draggable);
-	bool is_draggable();
+	int get_port_input_count();
+	Vector2 get_port_input_position(int p_port_idx);
+	int get_port_input_type(int p_port_idx);
+	Color get_port_input_color(int p_port_idx);
+	int get_port_input_slot(int p_port_idx);
```

Comment:
I think it would be better to call these `input/output_port` rather than `port_input/output`. For example, the name `get_port_input_count` doesn't read the way it should for what it returns. It reads like it returns some input count.

With the old names using "connection" it kind of worked, because it was "connection input". Where as with "port" the word "input" it its qualifier. In other words, connection input is input port.

Current format also adds to the whole slot/port confusion, I think. When you read "port input color", it feels like what "slot input color" would mean. An input color of something, implying there is also an output color of the same thing. Whereas "input port color" is more precise and less misleading.